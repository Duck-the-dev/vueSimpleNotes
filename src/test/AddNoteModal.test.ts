import AddNoteModalVue from '../components/AddNoteModal.vue'
import NoteItemVue from '../components/NoteItem.vue'
import { mount } from '@vue/test-utils'

describe('note', () => {
  test('add new note', async () => {
    expect(AddNoteModalVue).toBeTruthy()
    expect(NoteItemVue).toBeTruthy()
    let noteMsg = ''

    const wrapperNewNote = mount(AddNoteModalVue)
    noteMsg = 'feed the cat early in the morning'

    const note = mount(NoteItemVue, {
      props: {
        text: noteMsg,
      },
    })

    expect(wrapperNewNote.text()).toContain('Add note')
    expect(wrapperNewNote.html()).toMatchSnapshot()

    await wrapperNewNote.get('#addNoteModal').trigger('click')

    const input = wrapperNewNote.find('textarea')

    await input.setValue('feed the cat early in the morning')
    console.log(input.element.value)

    expect(input.element.value).toBe('feed the cat early in the morning')

    await wrapperNewNote.get('#addNoteConfirm').trigger('click')

    expect(note.text()).toBe('feed the cat early in the morning')
  }),
    test('add two notes', async () => {
      expect(AddNoteModalVue).toBeTruthy()
      expect(NoteItemVue).toBeTruthy()
      let noteMsg = ''

      const wrapperNewNote = mount(AddNoteModalVue)
      noteMsg = 'feed the cat early in the morning'

      const note1 = mount(NoteItemVue, {
        props: {
          text: noteMsg,
        },
      })

      expect(wrapperNewNote.text()).toContain('Add note')
      expect(wrapperNewNote.html()).toMatchSnapshot()

      await wrapperNewNote.get('#addNoteModal').trigger('click')

      const input = wrapperNewNote.find('textarea')

      await input.setValue('feed the cat early in the morning')
      console.log(input.element.value)

      expect(input.element.value).toBe('feed the cat early in the morning')

      await wrapperNewNote.get('#addNoteConfirm').trigger('click')

      expect(note1.text()).toBe('feed the cat early in the morning')

      noteMsg = 'feed the dog early in the morning'

      const note2 = mount(NoteItemVue, {
        props: {
          text: noteMsg,
        },
      })

      await wrapperNewNote.get('#addNoteModal').trigger('click')

      const input2 = wrapperNewNote.find('textarea')

      await input2.setValue(noteMsg)
      console.log(input.element.value)

      expect(input2.element.value).toBe(noteMsg)

      await wrapperNewNote.get('#addNoteConfirm').trigger('click')

      expect(note2.text()).toBe(noteMsg)
    })
})
