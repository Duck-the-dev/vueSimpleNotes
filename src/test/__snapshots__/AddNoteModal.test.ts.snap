// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`note > add new note 1`] = `
"<div class=\\"container mx-auto mt-5 mb-5\\">
  <!-- The button to open modal --><label id=\\"addNoteModal\\" for=\\"my-modal-6\\" class=\\"btn-xl btn-primary btn\\">Add note</label><!-- Put this part before </body> tag --><input type=\\"checkbox\\" id=\\"my-modal-6\\" class=\\"modal-toggle\\">
  <div class=\\"modal modal-bottom sm:modal-middle\\">
    <div class=\\"modal-box\\"><label for=\\"my-modal-6\\" class=\\"btn-sm btn-circle btn absolute right-2 top-2\\">✕</label>
      <div class=\\"form-control\\"><label class=\\"label\\"><span class=\\"label-text\\">Add New Note</span></label><textarea class=\\"textarea-bordered textarea h-24 max-h-52\\" placeholder=\\"Note content\\"></textarea></div>
      <div class=\\"modal-action\\" style=\\"display: none;\\"><label for=\\"my-modal-6\\" class=\\"btn\\" id=\\"addNoteConfirm\\">Add Note!</label></div>
    </div>
  </div>
</div>
<!-- grid of notes -->
<div class=\\"grid grid-flow-row gap-4 sm:grid-cols-1 md:grid-cols-4\\"></div>"
`;

exports[`note > add two notes 1`] = `
"<div class=\\"container mx-auto mt-5 mb-5\\">
  <!-- The button to open modal --><label id=\\"addNoteModal\\" for=\\"my-modal-6\\" class=\\"btn-xl btn-primary btn\\">Add note</label><!-- Put this part before </body> tag --><input type=\\"checkbox\\" id=\\"my-modal-6\\" class=\\"modal-toggle\\">
  <div class=\\"modal modal-bottom sm:modal-middle\\">
    <div class=\\"modal-box\\"><label for=\\"my-modal-6\\" class=\\"btn-sm btn-circle btn absolute right-2 top-2\\">✕</label>
      <div class=\\"form-control\\"><label class=\\"label\\"><span class=\\"label-text\\">Add New Note</span></label><textarea class=\\"textarea-bordered textarea h-24 max-h-52\\" placeholder=\\"Note content\\"></textarea></div>
      <div class=\\"modal-action\\" style=\\"display: none;\\"><label for=\\"my-modal-6\\" class=\\"btn\\" id=\\"addNoteConfirm\\">Add Note!</label></div>
    </div>
  </div>
</div>
<!-- grid of notes -->
<div class=\\"grid grid-flow-row gap-4 sm:grid-cols-1 md:grid-cols-4\\"></div>"
`;
